cloud-gov/cg-ui
# Functional users list overlays

Blocked by PR #476 

## Changes proposed in this pull request:

- Converts roles functionality in users list to overlays

Per [engineering sync discussion](https://github.com/cloud-gov/cg-ui/pull/476#issuecomment-2341945567), some additional acceptance criteria:

- [x] When opening the dialog, we'll keep focus on the close button and allow users to tab through the rest of the content.
- [x] Opening the dialog should also announce a title specific to the functionality being presented in the dialog.
- [x] When closing the dialog, we'll keep the default behavior of returning focus to the last point of regard.
- [ ] When closing the dialog, we'll have SR's announce any success messages (while keeping default focus).
- [x] While dialog is open, we agreed that a user should be able to click outside of it to close it. Since this is not default behavior for dialogs, we're going to spike on this and see what we come up with.

### How to test

1. Log into the CloudFoundry dev environment through the CLI: `cf login -a [url] --sso`
2. Start up a dev server with a refreshed CF token: `npm run dev-cf`
3. Navigate to the users list: `/orgs/[orgId]`
4. For the org roles form, click any link under the "organization roles" column
5. For the space roles form, click any link under the "access permissions" column

Forms should operate the same as when they were on pages, with the only difference being that the overlay closes on form submission success. 

### Related issues

Closes #468

### Submitter checklist


## Security considerations

None, UI changes only
